File: src/runtime.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void printInt(int i)
{
   printf("%d\n", i);
}


void printString(char* s)
{
   printf("%s\n", s);
}

char* readString(void)
{
   char *ptr=NULL;
   size_t len;
   
   getline(&ptr, &len, stdin);
   
   return ptr;
}

int readInt(void)
{
   char *ptr=NULL;
   size_t len;
   int out;
   
   getline(&ptr, &len, stdin);
   out = atoi(ptr);
   free(ptr);
   return out;
}
  
void error(void)
{
   printf("runtime error\n");
}

char* __concat(char* a, char*b)
{
   char* out;
   out=malloc(strlen(a)+strlen(b)+1);
   strcpy(out,a);
   strcat(out,b);
   
   return out;
}

File: lib/Runtime.java
package lib;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Runtime {
	static InputStreamReader converter;
	static BufferedReader in;
	
	public static String readString() throws IOException {
		if(converter==null){
			converter = new InputStreamReader(System.in);
			in = new BufferedReader(converter);
		}
		
		return in.readLine();
	}
	
	public static int readInt() throws IOException {
		String line=readString();
		
		return Integer.parseInt(line);
	}
	
	public static void printString(String s) {
		System.out.println(s);
	}

	public static void printInt(int i) {
		System.out.println(i);
	}

	public static void error(){
		System.exit(1);
	}
}
